src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java

```diff
@@ -239,7 +239,7 @@ private Pageable mapSortParameters(Pageable pageable) {
 
               return Mono.just(bookingTO);
             })
-        .transform(createShipmentEventFromBookingTO);
+            .flatMap(bTO -> createShipmentEventFromBookingTO(bTO).thenReturn(bTO));
```

Comment:
FTR, I think we can unwrap a level of lambdas via this change:

```suggestion
            .flatMap(this::createShipmentEventFromBookingTO)
            .thenReturn(bTO);
```
Either way works for me though.

Reply:
I like the suggestion pretty neat way to handle it, however bTO is formed within the chain flow so its not available outside of it. With the current chain of operations I can't use it, just tried it.

Reply:
Right, the `thenReturn` would fail.  This would only work if `createShipmentEventFromBookingTO` returned `bTO` again in which case the ´.thenReturn(bTO)` is redundant.